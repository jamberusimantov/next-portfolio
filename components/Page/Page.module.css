/* PageHeadWithImage */

.navSection {
    background-color: var(--blue);
    color: var(--white);
}

.navSectionContainer {
    display: flex;
    max-width: 1200px;
    height: 57px;
    margin: auto;
    padding: .5rem;
}

.nav {
    font-weight: 600;
    font-size: large;
    display: flex;
    width: 100%;
    flex-direction: row;
    align-items: center;
    min-width: max-content;
}

.breadcrumbs {
    display: flex;
    justify-self: flex-start;
}

.menu {
    display: flex;
    width: 100%;
    justify-content: flex-end;
}

.menuList {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin: 0;
    padding: 0;
    list-style: none;
    position: absolute;
    top: 57px;
    left: 0;
    width: 100%;
    background-color: var(--blue);
    z-index: -1;
}

.menuList.active {
    z-index: 1;
}

.menuList li {
    padding: .5rem;
    display: flex;
    justify-content: flex-start;
}

.hr {
    width: calc(100% - 1rem);
    margin: 0 .5rem;
}

.imageSection {
    height: 24rem;
    padding: 3rem 1rem 6rem;
    background-color: #333;
    background-image: url('/assets/bg/bg.jpg');
    background-position: center;
    background-size: cover;
}

@media(min-width:768px) {
    .menuList {
        flex-direction: row;
        z-index: 1;
        position: relative;
        top: initial;
        justify-content: flex-end;
    }
    .hr {
        display: none;
    }
}