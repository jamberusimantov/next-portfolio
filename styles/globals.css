html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}