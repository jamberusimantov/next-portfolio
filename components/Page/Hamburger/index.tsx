import React from 'react';
import styles from './hamburger.module.css';


interface iHamburgerProps {
    isMenuOpen: boolean,
    setIsMenuOpen: React.Dispatch<React.SetStateAction<boolean>>
};

const Hamburger = (props: iHamburgerProps) => {
    const { isMenuOpen, setIsMenuOpen } = props;
    const toggleMenu = () => {
        setIsMenuOpen(!isMenuOpen)
        document.querySelector(`button.${styles.hamburger}`)?.classList.toggle(styles.active);
    };
    return (
        <div className={styles.menuBtn}>
            <button className={styles.hamburger} onClick={toggleMenu}>
                <div className={styles.inner}>
                    <span className={styles.bar}></span>
                    <span className={styles.bar}></span>
                    <span className={styles.bar}></span>
                </div>
            </button>
        </div>
    );
};

export default Hamburger;