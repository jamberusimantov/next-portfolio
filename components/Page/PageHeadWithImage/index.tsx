import React, { useState, useContext } from 'react';
import styles from './PageHeadWithImage.module.css'
import CustomLink from '../../CustomLink';
import dictionary from '../../../dir/dictionary'
import { HTMLContext } from '../../../pages/_app'
// import { getAppStateHtml } from '../../../store/hooks'
import Hamburger from './Hamburger';


export interface iPageHeadWithImageProps {
    pageName?: string,
    imageClassName?: string,
}


const PageHeadWithImage = (props: iPageHeadWithImageProps) => {
    const { pageName, imageClassName } = props;
    const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false)
    // const html = getAppStateHtml();
    const { html } = useContext(HTMLContext);
    return (
        <>
            <div className={styles.navSection}>
                <div className={styles.navSectionContainer}>
                    <div className={styles.breadcrumbs}>
                        <CustomLink
                            href={'/'}
                            title={dictionary.homepage[html.lang]}
                            children={<span
                                className={"linkItem white"}
                                children={dictionary.homepage[html.lang]}
                            />}
                        />
                        {pageName && <CustomLink
                            href={`/${pageName}`}
                            title={dictionary[pageName][html.lang]}
                            children={[
                                <span className={"decoration"} key={0}>
                                    &gt;
                                </span>,
                                <span className={"linkItem white"} key={1}>
                                    {dictionary[pageName][html.lang]}
                                </span>
                            ]}
                        />}
                    </div>
                    {!pageName && <>
                        <Hamburger isMenuOpen={isMenuOpen} setIsMenuOpen={setIsMenuOpen} />
                        <ul className={`${styles.nav} ${isMenuOpen ? styles.active : ''}`}>
                            <li><CustomLink
                                href={'/profile'}
                                title={dictionary.profile[html.lang]}
                                children={<span
                                    className={"linkItem white"}
                                    children={dictionary.profile[html.lang]}
                                />}
                            /></li>
                            <hr className={styles.hr} />
                            <li><CustomLink
                                href={'/projects'}
                                title={dictionary.projects[html.lang]}
                                children={<span
                                    className={"linkItem white"}
                                    children={dictionary.projects[html.lang]}
                                />}
                            /></li>
                            <hr className={styles.hr} />
                            <li><CustomLink
                                href={'/contact'}
                                title={dictionary.contact[html.lang]}
                                children={<span
                                    className={"linkItem white"}
                                    children={dictionary.contact[html.lang]}
                                />}
                            /></li>
                            <hr className={styles.hr} />
                        </ul>
                    </>}
                </div>
            </div>
            {imageClassName && <div className={`${styles.imageSection} ${props.imageClassName}`}></div>}
        </>
    )
}

export default PageHeadWithImage;