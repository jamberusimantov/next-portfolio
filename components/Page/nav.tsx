import React, { useState, useContext } from 'react';
import styles from './Page.module.css';
import CustomLink from '../CustomLink';
import dictionary from '../../dir/dictionary';
import HTMLContext from '../../contexts/htmlContext';
import Hamburger from './Hamburger';


export const Menu = () => {
    const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
    const { html } = useContext(HTMLContext);
    return (
        <div className={styles.menu}>
            <Hamburger
                isMenuOpen={isMenuOpen}
                setIsMenuOpen={setIsMenuOpen}
            />
            <ul className={`${styles.menuList} ${isMenuOpen ? styles.active : ''}`}>
                <li>
                    <CustomLink
                        href={'/profile'}
                        title={dictionary.profile["en-US"]}
                        children={dictionary.profile[html.lang]}
                        className="white"
                    />
                </li>
                <hr className={styles.hr} />
                <li>
                    <CustomLink
                        href={'/projects'}
                        title={dictionary.projects["en-US"]}
                        children={dictionary.projects[html.lang]}
                        className="white"
                    />
                </li>
                <hr className={styles.hr} />
                <li>
                    <CustomLink
                        href={'/contact'}
                        title={dictionary.contact["en-US"]}
                        children={dictionary.contact[html.lang]}
                        className="white"
                    />
                </li>
                <hr className={styles.hr} />
            </ul>
        </div>
    );
};

const Decoration = () => <span className={"decoration"}>&gt;</span>;

export const Breadcrumbs = (props: { pageName: string }) => {
    const { html } = useContext(HTMLContext);
    return (
        <div className={styles.breadcrumbs}>
            <CustomLink
                href={'/'}
                title={"Homepage"}
                children={dictionary.homepage[html.lang]}
                className="white"
            />
            <Decoration />
            <CustomLink
                href={`/${props.pageName}`}
                disabled
                children={dictionary[props.pageName][html.lang]}
            />
        </div>
    );
};