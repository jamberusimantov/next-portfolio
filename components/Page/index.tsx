import React, { ReactElement } from 'react';
import styles from './Page.module.css'; 
import { Menu, Breadcrumbs } from './nav';


interface iPageProps {
    homepage?: boolean,
    pageName?: string,
    imageClassName?: string,
    children?: ReactElement,
};

const Page = (props: iPageProps) => {
    return (
        <div className="page">
            <div className={!props.homepage ? "pageContainer" : ""}>
                <div className={styles.navSection}>
                    <div className={styles.navSectionContainer}>
                        <div className={styles.nav}>
                            {props.pageName ? <Breadcrumbs pageName={props.pageName} /> : <Menu />}
                        </div>
                    </div>
                </div>
                {props.imageClassName && <div className={`${styles.imageSection} ${props.imageClassName}`}></div>}
                {props.children}
            </div>
        </div>
    );
}

export default Page;