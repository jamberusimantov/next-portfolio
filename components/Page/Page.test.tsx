/**
 * @jest-environment jsdom
 */
import React from "react";
import { describe, expect, test } from '@jest/globals';
import { render, act } from '@testing-library/react';
import Page from '.';


describe('Page', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        await act(async () => {
            let { container } = await render(<Page />);
            el = container;
        });
        expect(el).toMatchSnapshot();
    });
});