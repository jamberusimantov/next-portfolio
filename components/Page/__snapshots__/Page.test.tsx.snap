// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="pageContainer"
    >
      <div
        class="navSection"
      >
        <div
          class="navSectionContainer"
        >
          <div
            class="nav"
          >
            <div
              class="menu"
            >
              <div
                class="menuBtn"
              >
                <button
                  class="hamburger"
                >
                  <div
                    class="inner"
                  >
                    <span
                      class="bar"
                    />
                    <span
                      class="bar"
                    />
                    <span
                      class="bar"
                    />
                  </div>
                </button>
              </div>
              <ul
                class="menuList "
              >
                <li>
                  <a
                    class="white"
                    href="/profile"
                    title="Profile"
                  >
                    פרופיל
                  </a>
                </li>
                <hr
                  class="hr"
                />
                <li>
                  <a
                    class="white"
                    href="/projects"
                    title="Projects"
                  >
                    עבודות
                  </a>
                </li>
                <hr
                  class="hr"
                />
                <li>
                  <a
                    class="white"
                    href="/contact"
                    title="Contact"
                  >
                    צור קשר
                  </a>
                </li>
                <hr
                  class="hr"
                />
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
